amlip_docs/rst/demo/wan.rst

```diff
@@ -0,0 +1,189 @@
+with different :term:`APIs <API>`, and to prove the communication between the 2 implementations.
+
+Simulation
+==========
+
+TensorFlow
+----------
+
+TensorFLow is an end-to-end machine learning platform with pre-trained models.
+
+Darknet
+-------
+
+Darknet is an open source neural network framework written in C and CUDA with pre-trained models.
+
+Edge Node
+---------
+
+This node simulates a :ref:`user_manual_nodes_edge`.
+It is implemented in |python| using :code:`amlip_py` API.
+There are 2 different ways to run it, an automatic one and a manual one:
+
+Inference Node
+--------------
+
+This node simulates a :ref:`user_manual_nodes_inference`.
+It is implemented in |cpp| using :code:`amlip_cpp` API.
+
+To run it, one string argument is required. This will be the model this node will use to make the inference.
+
+Installation
+============
+
+First of all, check that :code:`amlip_demo_nodes` sub-package is correctly installed.
+If it is not, please refer to :ref:`developer_manual_installation_sources_linux_colcon_demos`.
+
+Requirements
+============
+
+The demo requires the following tools to be installed in the system:
+
+.. code-block:: bash
+
+    sudo apt install -y  swig espeak alsa-utils libopencv-dev ffmpeg
+    pip3 install -U pyttsx3 opencv-python tensorflow tensorflow-hub tensorflow-object-detection-api pygame
+
+Get the models
+==============
+
+Using TensorFlow
+----------------
+
+.. code-block:: bash
+    cd ~/AML-IP-demos-ws/src/AML-IP-demos/amlip_demos/resource/tensorflow/models/
+    wget -O centernet_hourglass_512x512_kpts_1.tar.gz https://tfhub.dev/tensorflow/centernet/hourglass_512x512_kpts/1?tf-hub-format=compressed
+    mkdir centernet_hourglass_512x512_kpts_1
+    tar -xvf centernet_hourglass_512x512_kpts_1.tar.gz -C ./centernet_hourglass_512x512_kpts_1
+
+Using Darknet
+-------------
+
+.. code-block:: bash
+    cd ~/AML-IP-demos-ws/src/AML-IP-demos/amlip_demos/resource/darknet/
+    wget https://pjreddie.com/media/files/yolov3-tiny.weights
+
+Run demo
+========
+
+The demo that is presented here follows the schema of the figure below:
+
+.. figure:: /rst/figures/workload_distribution_basic_demo.png
+    :width: 50%
+
+Run Edge Node
+-------------
+
+Run the following command:
+
+.. code-block:: bash
+
+    # Source colcon installation
+    source install/setup.bash
+
+    # To execute Edge Node to send an image to inferred
+    cd ~/AML-IP-demos-ws/src/AML-IP-demos/amlip_demos/amlip_demos
+    python3 edge_node.py
+
+Take into account that this node will wait until there is an *Inference Node* running and available
+in the same :term:`LAN` in order to process the inference.
+The expected output is the following:
+
+.. code-block:: bash
+
+    Edge Node AMLEdgeNode.e5.45.c7.f5 ready.
+    Edge Node AMLEdgeNode.e5.45.c7.f5 sending data: /home/irenebm/annapurna/AML-IP/src/amlipdemos/amlip_demos/resource/tensorflow/models/research/object_detection/test_images/image2.jpg.
```

Comment:
If you are using specific images, you should speak about them and make them public here for the user to try